payment-request/interfaces.https.html

```diff
@@ -5,136 +5,26 @@
-};
-dictionary PaymentItem {
-  required DOMString label;
-  required PaymentCurrencyAmount amount;
-  boolean pending = false;
-};
-[SecureContext]
-interface PaymentAddress {
-  [Default] object toJSON();
-  readonly attribute DOMString country;
-  readonly attribute FrozenArray<DOMString> addressLine;
-  readonly attribute DOMString region;
-  readonly attribute DOMString city;
-  readonly attribute DOMString dependentLocality;
-  readonly attribute DOMString postalCode;
-  readonly attribute DOMString sortingCode;
-  readonly attribute DOMString languageCode;
-  readonly attribute DOMString organization;
-  readonly attribute DOMString recipient;
-  readonly attribute DOMString phone;
-};
-dictionary PaymentShippingOption {
-  required DOMString id;
-  required DOMString label;
-  required PaymentCurrencyAmount amount;
-  boolean selected = false;
-};
-enum PaymentComplete {
-  "fail",
-  "success",
-  "unknown"
-};
-[SecureContext]
-interface PaymentResponse {
-  [Default] object toJSON();
-
-  readonly attribute DOMString requestId;
-  readonly attribute DOMString methodName;
-  readonly attribute object details;
-  readonly attribute PaymentAddress? shippingAddress;
-  readonly attribute DOMString? shippingOption;
-  readonly attribute DOMString? payerName;
-  readonly attribute DOMString? payerEmail;
-  readonly attribute DOMString? payerPhone;
-
-  Promise<void> complete(optional PaymentComplete result = "unknown");
-};
-[Constructor(DOMString type, optional PaymentRequestUpdateEventInit eventInitDict),
-  SecureContext]
-interface PaymentRequestUpdateEvent : Event {
-  void updateWith(Promise<PaymentDetailsUpdate> detailsPromise);
-};
-dictionary PaymentRequestUpdateEventInit : EventInit {
-};
-</script>
-
 <script>
-"use strict";
-var idlArray = new IdlArray();
-[].forEach.call(document.querySelectorAll("script[type=text\\/plain]"), function(node) {
-  if (node.className == "untested") {
-    idlArray.add_untested_idls(node.textContent);
-  } else {
-    idlArray.add_idls(node.textContent);
-  }
-});
-idlArray.add_objects({
-  PaymentRequest: ["new PaymentRequest([{supportedMethods: 'foo'}], {total: {label: 'bar', amount: {currency: 'BAZ', value: '0'}}})"]
-});
-idlArray.test();
+promise_test(async () => {
+  const idlArray = new IdlArray();
+  const untestedIDL = `
+    interface EventHandler {};
+    interface Event {};
+    interface EventInit {};
+    interface EventTarget {};
+  `;
+  idlArray.add_untested_idls(untestedIDL);
+  const testedIdl = await fetch("/interfaces/payment-request.idl").then(r =>
+    r.text()
+  );
+  idlArray.add_idls(testedIdl);
+  const methods = "[{supportedMethods: 'foo'}]";
+  const amount = "{currency: 'USD', value: '0'}";
+  const details = `{total: {label: 'bar', amount: ${amount} }}`;
+  idlArray.add_objects({
+    PaymentRequest: [`new PaymentRequest(${methods}, ${details})`],
+  });
+  idlArray.test();
+}, "Must have a conforming IDL implementation of the Payment Request API.");
```

Comment:
I don't think if this test passes that will be true. The test will just pass if the fetch succeeded, etc. So I would label this "Setup for Payment Request API IDL tests" or similar.